src/main/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervals.java

```diff
@@ -0,0 +1,128 @@
+ * intervals (.list or .intervals), BED files and VCF files. If no intervals are given, each contig will be covered
+ * by a single interval (whole genome sequencing). </p>
+ *
+ * <p>Using the -P flag, the user can specify the amount of padding (in bp) added to each side of the intervals.
+ * This padding is in addition to the padding added by the -ip (or --interval_padding) argument of IntervalArgumentCollection.
+ * We encourage using only the -P flag.
+ *
+ * <p>The user can also specify the length of the bins (in bp) using the -BL option. If this is not commensurate with
+ * the length of the padded intervals, then the last bin will be of different size than the others. </p>
+ *
+ * <h3>Example</h3>
+ * <pre>
+ * gatk-launch --javaOptions "-Xmx4g" PreprocessIntervals \
+ *   -R ref_fasta.fa \
+ *   -L intervals.list \
+ *   -BL 10000 \
+ *   -P 500 \
+ *   -O interval-files
+ * </pre>
+ *
+ * @author Marton Kanasz-Nagy &lt;mkanaszn@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        summary = "Preprocess intervals given by the user and then create bins that cover them. "
+                + "The intervals are first padded, the overlapping ones are merged and bins are created that cover the intervals. "
+                + "If the bin size is incommensurate with the length of the interval, the last bin of the interval might be of different size than the the rest of the bins."
+                + "The length of the padding regions at both sides of the intervals (-P) and the length of the bins (-BL) can be given as command line arguments. ",
+        oneLineSummary = "Preprocess intervals given by the user and then create bins that cover them.",
+        programGroup = CopyNumberProgramGroup.class
+)
+@DocumentedFeature
+public final class PreprocessIntervals extends GATKTool {
+    public static final String LENGTH_OF_BINS_SHORT_NAME = "BL";
+    public static final String LENGTH_OF_BINS_LONG_NAME = "binLength";
+
+    public static final String PADDING_SHORT_NAME = "P";
+    public static final String PADDING_LONG_NAME = "padding";
+
+    @Argument(
+            doc = "Length (in bp) of the bins.",
+            fullName = LENGTH_OF_BINS_LONG_NAME,
+            shortName = LENGTH_OF_BINS_SHORT_NAME,
+            optional = true,
+            minValue = 1
+    )
+    private int binLength = 1000;
+
+    @Argument(
+            doc = "Length (in bp) of the padding regions on each side of the intervals.",
+            fullName = PADDING_LONG_NAME,
+            shortName = PADDING_SHORT_NAME,
+            optional = true,
+            minValue = 0
+    )
+    private int padding = 0;
+
+    @Argument(
+            doc = "Output file.",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+    )
+    private File outputFile;
+
+    @Override
+    public void onTraversalStart() {
+        createBins();
+    }
+
+    private void createBins() {
```

Comment:
This method generates *and* writes the bins, so perhaps just `generateAndWriteBins()` would be better.  This also distinguishes it from `generateBins()` below (otherwise, what is the difference between "creating" and "generating"?)

Reply:
Great point! I changed it to 'prepareIntervalsGenerateAndWriteBins()' which is more specific to what the method does.